.github/workflows/e2e.yml

```diff
@@ -0,0 +1,219 @@
+        echo "DEST_REPORTS_DIR=$REPORT_PREFIX/$REPORT_DATE/${{ github.run_id }}/${{ github.run_attempt }}" >> $GITHUB_ENV
+
+    - name: Upload fixtures reports
+      uses: actions/upload-artifact@v4
+      with:
+          name: testplane-fixtures
+          path: test/func/fixtures/*/report
+
+    - name: 'e2e: Run Testplane testplane-common'
+      id: 'testplane-testplane-common'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:testplane-common
+
+    - name: 'e2e: Run Testplane testplane-eye'
+      id: 'testplane-testplane-eye'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:testplane-eye
+
+    - name: 'e2e: Run Testplane testplane-gui'
+      id: 'testplane-testplane-gui'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:testplane-gui
+
+    - name: 'e2e: Run Testplane playwright'
+      id: 'testplane-playwright'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:playwright
+
+    - name: 'e2e: Run Testplane plugins'
+      id: 'testplane-plugins'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:plugins
+
+    - name: 'e2e: Run Testplane testplane-tinder'
+      id: 'testplane-testplane-tinder'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:testplane-tinder
+
+    - name: 'e2e: Run Testplane analytics'
+      id: 'testplane-analytics'
+      continue-on-error: true
+      working-directory: 'test/func/tests'
+      run: npm run testplane:analytics
+
+    - name: 'e2e: Stop browser docker image'
+      run: |
+        docker kill ${{ env.DOCKER_IMAGE_NAME }} || true
+        docker rm ${{ env.DOCKER_IMAGE_NAME }} || true
+
+    - name: Set IS_FAILED if any step didn't succeed
+      if: always()
+      run: |
+        if [ "${{ steps.testplane-testplane-common.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-testplane-gui.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-plugins.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-analytics.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-playwright.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-testplane-eye.outcome }}" != "success" ] || \
+            [ "${{ steps.testplane-testplane-tinder.outcome }}" != "success" ]; then
+          echo "IS_FAILED=true" >> $GITHUB_ENV
+        fi
+
+    - name: Deploy Testplane html-reporter reports
+      if: ${{ env.IS_FAILED }}
+      uses: peaceiris/actions-gh-pages@v4
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ${{ env.SOURCE_REPORTS_DIR }}
+        destination_dir: ${{ env.DEST_REPORTS_DIR }}
+        keep_files: true
+
+    - name: Construct failed PR comment
```

Comment:
Couldn't figure out smarter way to do this.
I tried constructing variables like `STEP_TESTPLANE_TESTPLANE_COMMON_OUTCOME` and evaluating them, but it does not work